packages/myst-transforms/src/enumerate.ts

```diff
@@ -323,15 +328,31 @@ export class ReferenceState implements IReferenceState {
   incrementCount(node: TargetNodes, kind: TargetKind | string): string {
     if (kind === TargetKind.heading && node.type === 'heading') {
       // Ideally initializeNumberedHeadingDepths is called before incrementing
-      // heading count to do a better job initializng headers based on tree
+      // heading count to do a better job initializing headers based on tree
       if (!this.targetCounts.heading) this.targetCounts.heading = [0, 0, 0, 0, 0, 0];
       this.targetCounts.heading = incrementHeadingCounts(node.depth, this.targetCounts.heading);
       return formatHeadingEnumerator(this.targetCounts.heading, this.numbering.enumerator);
     }
-    if (kind in this.targetCounts) {
-      this.targetCounts[kind] += 1;
-    } else {
-      this.targetCounts[kind] = 1;
+    if (kind === TargetKind.subequation && node.type === 'math') {
+      // This should be reset every time a mathGroup is encountered (see below)
+      this.targetCounts.subequation ??= [this.targetCounts.equation, 0];
+      this.targetCounts.subequation = [
+        this.targetCounts.subequation[0],
+        this.targetCounts.subequation[1] + 1,
+      ];
+      const sub = this.targetCounts.subequation[1];
+      // Will restart counting if there are more than 26 subequations
+      const letter = String.fromCharCode(((sub - 1) % 26) + 'a'.charCodeAt(0));
+      const enumerator = this.targetCounts.subequation[0] + letter;
```

Comment:
These subequations always have sub-enumeration, i.e. there is no option to enumerate equations and subequations all sequentially. Seems like this is something we will want eventually. (And probably more complex than just frontmatter `numbering.subequation = false` - in case in a single paper you need both.)

This definitely seems fine for now though

Reply:
Yeah, we can also have align groups as `math` - and if we start transforming those into `mathGroups`, then we will want different numbering for each aligned equation.

Good enough for this pass though! :)